lib/srv/desktop/windows_server.go

```diff
@@ -799,27 +799,19 @@ func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger,
 		recordSession = recConfig.GetMode() != types.RecordOff
 	}
 
-	var windowsUser string
-	authorize := func(login string) error {
-		windowsUser = login // capture attempted login user
-		state := authCtx.GetAccessState(authPref)
-		return authCtx.Checker.CheckAccess(
-			desktop,
-			state,
-			services.NewWindowsLoginMatcher(login))
-	}
-
 	// Use a context that is canceled when we're done handling
 	// this connection. This ensures that the connection monitor
 	// will stop checking for idle activity when the connection
 	// is closed.
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	// Create a session tracker so that other services, such as
-	// the session upload completer, can track the session's lifetime.
-	if err := s.trackSession(ctx, &identity, windowsUser, string(sessionID), desktop); err != nil {
-		return trace.Wrap(err)
+	authorize := func(login string) error {
+		state := authCtx.GetAccessState(authPref)
```

Comment:
This callback is now purely authorization and no longer grabs the `windowsUser` varbiable - we get that from the client instead.